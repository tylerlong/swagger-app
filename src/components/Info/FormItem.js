import React from 'react'
import { connect } from 'react-redux'
import { Input, Form } from 'antd'

import { setProp } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component {
  render () {
    console.log(`render Info.FormItem`)
    const { name, value, setProp } = this.props
    return (
      <Form.Item {...formItemLayout} label={name}>
        <Input placeholder={name} size='large' value={value} onChange={(event) => { setProp(name, event.target.value) }} />
      </Form.Item>
    )
  }
}

const mapStateToProps = ({ info }, { name }) => ({ value: info[name] })
const mapDispatchToProps = (dispatch) => ({
  setProp: (key, value) => dispatch(setProp(['info', key], value))
})
export default connect(mapStateToProps, mapDispatchToProps)(FormItem)
