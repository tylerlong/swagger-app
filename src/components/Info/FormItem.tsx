import React from 'react'
import R from 'ramda'
import { connect } from 'react-redux'
import { Input, Form } from 'antd'

import { setProp } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component<any> {
  render() {
    console.log(`render Info.FormItem`)
    const { name, value, setProp } = this.props
    let input = <Input placeholder={name} size='large' value={value} onChange={(event: any) => { setProp(name, event.target.value) }} />
    if (R.contains(name, ['schemes', 'produces', 'consumes'])) {
      input = <Input placeholder='Values separated by commas' size='large' defaultValue={R.join(', ', value)} onChange={(event: any) => { setProp(name, R.pipe(R.split(','), R.map(R.trim), R.reject(R.equals('')))(event.target.value)) }} />
    }
    return (
      <Form.Item {...formItemLayout} label={name}>
        {input}
      </Form.Item>
    )
  }
}

const mapStateToProps = ({ info }, { name }) => ({ value: info[name] })
const mapDispatchToProps = (dispatch) => ({
  setProp: (key, value) => dispatch(setProp(['info', key], value))
})
export default connect(mapStateToProps, mapDispatchToProps)(FormItem)
