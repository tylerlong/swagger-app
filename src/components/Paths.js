import React from 'react'
import R from 'ramda'
import { Button } from 'antd'
import { connect } from 'react-redux'

import { addPath } from '../actions'

class Paths extends React.Component {
  render () {
    return (
      <div>
        <h2>Paths</h2>
        <div className='primary-button'>
          <Button type='primary' size='large' onClick={this.props.addPath}>Add path</Button>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { addPath })(Paths)
