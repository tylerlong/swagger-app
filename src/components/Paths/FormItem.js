import React from 'react'
import { connect } from 'react-redux'
import R from 'ramda'
import { Popconfirm, Button, Icon, Form, Input } from 'antd'

import { setProp, deletePath } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component {
  render () {
    const { index, paths, setProp, deletePath } = this.props
    const path = paths[index]
    if (!path) {
      return null
    }
    return (
      <div>
        <Popconfirm title={`Are you sure to delete path "${path.path}"?`} okText='Yes' cancelText='No' onConfirm={() => deletePath(index)}>
          <Button type='danger'><Icon type='arrow-up' />Delete</Button>
        </Popconfirm>
        <Form.Item {...formItemLayout} label='Name'>
          <Input placeholder='Path' size='large' value={path.path} onChange={(event) => { setProp(['paths', index, 'path'], event.target.value) }} />
        </Form.Item>
      </div>
    )
  }
}

export default connect(R.identity, { setProp, deletePath })(FormItem)
