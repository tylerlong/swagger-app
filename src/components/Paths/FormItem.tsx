import React from 'react'
import { connect } from 'react-redux'
import { Popconfirm, Button, Icon, Form, Input } from 'antd'

import { setProp, deletePath } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component<any> {
  render() {
    console.log(`render Paths.FormItem`)
    const { path, setProp, deletePath } = this.props
    if (!path) {
      return null
    }
    return (
      <div>
        <Popconfirm title={`Are you sure to delete path "${path.path}"?`} okText='Yes' cancelText='No' onConfirm={deletePath}>
          <Button type='danger'><Icon type='arrow-up' />Delete</Button>
        </Popconfirm>
        <Form.Item {...formItemLayout} label='Name'>
          <Input placeholder='Path' size='large' value={path.path} onChange={(event: any) => { setProp('path', event.target.value) }} />
        </Form.Item>
      </div>
    )
  }
}

const mapStateToProps = ({ paths }, { index }) => ({ path: paths[index] })
const mapDispatchToProps = (dispatch, { index }) => ({
  deletePath: () => dispatch(deletePath(index)),
  setProp: (key, value) => dispatch(setProp(['paths', index, key], value))
})
export default connect(mapStateToProps, mapDispatchToProps)(FormItem)
