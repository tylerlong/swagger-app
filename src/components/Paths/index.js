import React from 'react'
import R from 'ramda'
import { Button } from 'antd'
import { connect } from 'react-redux'

import { addPath } from '../../actions'

class Paths extends React.Component {
  render () {
    const { addPath } = this.props
    return (
      <div>
        <h2>Paths</h2>
        <div className='primary-button'>
          <Button type='primary' size='large' onClick={addPath}>Add path</Button>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { addPath })(Paths)
