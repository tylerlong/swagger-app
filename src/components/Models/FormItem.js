import React from 'react'
import R from 'ramda'
import { connect } from 'react-redux'
import { Input, Form, Button, Popconfirm, Collapse, Card, Row, Col } from 'antd'

import { setProp, deleteModel, addModelProperty } from '../../actions'
import { formItemLayout } from '../../utils/components'
import SubFormItem from './SubFormItem'

class FormItem extends React.Component {
  render () {
    const { index, models, setProp, deleteModel, addModelProperty } = this.props
    const model = models[index]
    return (
      <div>
        <Form.Item {...formItemLayout} label='Name'>
          <Input placeholder='Name' size='large' value={model.name} onChange={(event) => { setProp(['models', index, 'name'], event.target.value) }} />
        </Form.Item >
        <Row type='flex' justify='center'>
          <Col xs={24} sm={18}>
            <Card title='Properties'>
              <Collapse accordion>
                {model.properties.map((prop, index2) => {
                  return (
                    <Collapse.Panel header={prop.name} key={prop.createdAt}>
                      <SubFormItem index1={index} index2={index2} />
                    </Collapse.Panel>
                  )
                })}
              </Collapse>
              <div className='primary-button'>
                <Button type='primary' size='large' onClick={() => addModelProperty(index)}>Add property</Button>
              </div>
            </Card>
          </Col>
        </Row>
        <div className='button-line'>
          <Popconfirm title={`Are you sure to delete model "${model.name}"?`} okText='Yes' cancelText='No' onConfirm={() => deleteModel(index)}>
            <Button type='danger'>Delete {model.name}</Button>
          </Popconfirm>
        </div>
      </div >
    )
  }
}

export default connect(R.identity, { setProp, deleteModel, addModelProperty })(FormItem)
