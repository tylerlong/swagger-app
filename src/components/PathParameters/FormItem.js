import React from 'react'
import R from 'ramda'
import { connect } from 'react-redux'
import { Input, Form, Button, Popconfirm } from 'antd'

import { setProp, deletePathParameter } from '../../actions'
import { formItemLayout } from '../../utils/components'

class FormItem extends React.Component {
  render () {
    const { index, pathParameters, setProp, deletePathParameter } = this.props
    const pathParameter = pathParameters[index]
    return (
      <div>
        <Form.Item {...formItemLayout} label='Name'>
          <Input placeholder='Name' size='large' value={pathParameter.name} onChange={(event) => { setProp(['pathParameters', index, 'name'], event.target.value) }} />
        </Form.Item>
        <Form.Item {...formItemLayout} label='Description'>
          <Input placeholder='Description' size='large' value={pathParameter.description} onChange={(event) => { setProp(['pathParameters', index, 'description'], event.target.value) }} />
        </Form.Item>
        <Form.Item {...formItemLayout} label='Enum'>
          <Input placeholder='Enum values separated by commas' size='large' defaultValue={R.join(', ', pathParameter.enum)} onChange={(event) => { setProp(['pathParameters', index, 'enum'], R.pipe(R.split(','), R.map(R.trim), R.reject(R.equals('')))(event.target.value)) }} />
        </Form.Item>
        <div className='button-line'>
          <Popconfirm title='Are you sure to delete it?' okText='Yes' cancelText='No' onConfirm={() => deletePathParameter(index)}>
            <Button type='danger'>Delete</Button>
          </Popconfirm>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { setProp, deletePathParameter })(FormItem)
