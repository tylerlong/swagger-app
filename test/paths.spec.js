/* eslint-env jest */
import R from 'ramda'
import { Popconfirm } from 'antd'

import Paths from '../src/web/components/Paths'
import FormItem from '../src/web/components/Paths/FormItem'
import { getWrapper } from './shared'
import state from '../dist/state.json'
import store from './store'

let wrapper = null
beforeEach(() => {
  wrapper = getWrapper(Paths, R.pick(['paths'], state))
})
const count = state.paths.length
const getCount = () => store.getState().paths.length
const getModel = (index) => store.getState().paths[index]

describe('test paths', () => {
  test('view paths', () => {
    expect(wrapper.find('.ant-collapse-item').length).toEqual(count)
  })

  test('add path', () => {
    wrapper.find('button.ant-btn-primary').simulate('click')
    expect(getCount()).toEqual(count + 1)
  })

  test('delete path', () => {
    wrapper.find('div.ant-collapse-header').first().simulate('click')
    wrapper.find(Popconfirm).props().onConfirm()
    expect(getCount()).toEqual(count - 1)
  })

  test('edit path path', () => {
    wrapper.find('div.ant-collapse-header').first().simulate('click')
    const index = wrapper.find(FormItem).first().props().index
    wrapper.find('input').first().simulate('change', { target: { value: '/hello' } })
    expect(getModel(index).path).toEqual('/hello')
    wrapper.find('input').first().simulate('change', { target: { value: '/world' } })
    expect(getModel(index).path).toEqual('/world')
  })
})
