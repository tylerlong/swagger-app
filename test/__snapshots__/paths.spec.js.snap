// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test paths view paths 1`] = `
<Provider
  store={
    Object {
      "actions": Array [],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Paths)>
    <Paths
      addPath={[Function]}
      paths={
        Array [
          Object {
            "createdAt": 1498875254374,
            "methods": Array [
              Object {
                "apiGroup": "Light",
                "batch": false,
                "description": "Get Server Info",
                "examples": Array [],
                "method": "get",
                "parameters": Object {},
                "permissions": Array [],
                "request": Object {},
                "response": Object {},
                "since": "1.0.2",
                "status": "normal",
                "tags": Array [
                  "API Versions",
                ],
                "visibility": "public",
              },
            ],
            "path": "/restapi",
          },
        ]
      }
    >
      <div>
        <h2>
          Paths
        </h2>
        <Collapse
          accordion={true}
          bordered={true}
          openAnimation={
            Object {
              "appear": [Function],
              "enter": [Function],
              "leave": [Function],
            }
          }
          prefixCls="ant-collapse"
        >
          <Collapse
            accordion={true}
            bordered={true}
            className=""
            destroyInactivePanel={false}
            onChange={[Function]}
            openAnimation={
              Object {
                "appear": [Function],
                "enter": [Function],
                "leave": [Function],
              }
            }
            prefixCls="ant-collapse"
          >
            <div
              className="ant-collapse"
            >
              <CollapsePanel
                destroyInactivePanel={false}
                header="/restapi"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    /restapi
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
            </div>
          </Collapse>
        </Collapse>
        <Button
          clicked={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          size="large"
          type="primary"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg"
            onClick={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <Icon
              type="plus"
            >
              <i
                className="anticon anticon-plus"
              />
            </Icon>
            <span>
              Add path
            </span>
          </button>
        </Button>
      </div>
    </Paths>
  </Connect(Paths)>
</Provider>
`;
